packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart

```diff
@@ -9,25 +9,19 @@ import 'package:vm_service/vm_service.dart';
 
 import '../../auto_dispose.dart';
 import '../../globals.dart';
+import 'debugger_service.dart';
 
 /// Responsible for managing the debug state of the app.
 class DebuggerController extends DisposableController
     with AutoDisposeControllerMixin {
   DebuggerController() {
     switchToIsolate(serviceManager.isolateManager.selectedIsolate);
-    autoDispose(serviceManager.isolateManager.onSelectedIsolateChanged
-        .listen(switchToIsolate));
-
-    autoDispose(_service.onDebugEvent.listen(_handleIsolateEvent));
+    autoDispose(
+        debuggerService.onSelectedIsolateChanged.listen(switchToIsolate));
+    autoDispose(debuggerService.onDebugEvent.listen(_handleIsolateEvent));
   }
 
-  // TODO(djshuckerow): Separate the controller from a debugging service object
-  // such that the controller does not need to know about VmService.
-  VmService get _service => serviceManager.service;
-  InstanceRef get reportedException => _reportedException;
-
-  IsolateRef isolateRef;
-  List<ScriptRef> scripts;
+  final debuggerService = DebuggerService();
```

Comment:
All parts of this refactor are great except for adding the debuggerService class. Lets double down on everyone working on this code being familiar with the VmService API and avoid writing abstractions on top of it when they don't add clear value.  I would leave the`VmService get _service => ...` getter and delete the debuggerService field.
If you want the DebuggerService class for mocking and testing then I could be convinced but  it would be good to refactor the API a little so it is clear what the sependencies of the DebuggerService are. Currently it is getting all its data behind the scenes from the serviceManager singleton.